test/integration/targets/blocks/72781.yml

```diff
@@ -0,0 +1,13 @@
+- hosts: all
+  gather_facts: no
+  any_errors_fatal: true
+  tasks:
+    - block:
```

Comment:
I tried adding an always to the outer block of one of the test cases, expecting that the only difference would be that both hosts execute the always task.

```yaml
- hosts: host1,host2
  gather_facts: no
  any_errors_fatal: true
  tasks:
    - block:
        - block:
            - fail:
              when: inventory_hostname == 'host1'
          rescue:
            - fail:
        - block:
            - debug: msg="SHOULD NOT HAPPEN"

      always:
        - debug: msg="SHOULD"
```

But host2 starts where it left off, and runs the debug that shouldn't happen before the `always`. I've been trying to spot the issue and cannot... :dizzy_face:

Reply:
That's right, I forgot that was an existing issue - 31543.